### ISSUE TITLE
ENOENT: no such file or directory while scaling and converting

### ISSUE THREAD

**Issue by tpfbnz** [upscayl/upscayl#450]
See log. Starts off well, notes correct input picture and output dir, but appears to want to process the output file as input but which does not exist at that point. Upscayl process does not appear to detect lack of valid input file, continues to show processing in progress, does not appear to time out.
Batch processing hung at getting rememberoutputfolder, I stopped process after a few minutes of inactivity., does not process pictures, creates wrongly named folder and stops. Fault reported in log.
I was using deb install in ubuntu setup. 
[upscayl-output2.txt](https://github.com/upscayl/upscayl.github.io/files/12521234/upscayl-output2.txt)

**Comment by NayamAmarshe** [comment]
```
🐞 BACKEND REPORTED:  ♻ Scaling and converting now...
🐞 BACKEND REPORTED:  ❌ Error reading original image metadata Error: ENOENT: no such file or directory, open '/home/tpf/Pictures/2/65853291_medium-243854751_upscayl_4x_realesrgan-x4plus.jpg'
```

This is the issue. It's not able to read the file. 

Could you send the image, if possible?

**Comment by tpfbnz** [comment]
![1zywt78l6pj91](https://github.com/upscayl/upscayl/assets/144107487/5928b676-bbe6-4e05-94f2-e16490553aa8)

Hi, this was one pic used to test issue
cheers
bob

**Comment by tpfbnz** [comment]
Pushed wrong button, new guy. :-)

**Comment by tpfbnz** [comment]
Hi, thanks for looking into this. For the sake of clarity, I just want to restate the problem as I was not as clear as I could have been.

- I installed the latest version, 2.7.5
- Set up options : default theme, jpg, 4x, 0% compression,output folder permanent(on), overwrite (off), GPU ID (blank)
- Upscayl - batch ( off)  first runs then (on) - note both failed to complete,
-  selected image, named '1zywt78l6pj91.jpg'
- selected output dir, named ~/Pictures/2
- Pushed Upscayl button -  Upscayl's rotate icon in picture frame worked for several minutes, used system monitoring to observe very little activity, stopped it manually
With the batch process, notes above probably sufficient.
Appears to be more job control issue than anything major. Previous version experience great, although I have only used it a time or two.

FYI :I am using Graphics:
  Device-1: AMD Navi 14 [Radeon RX 5500/5500M / Pro 5500M] driver: amdgpu
    v: kernel
  Display: wayland server: X.Org v: 1.22.1.8 with: Xwayland v: 22.1.8
    compositor: gnome-shell driver: X: loaded: amdgpu

- I seemed to need additional Vulkan support to use Upscayl so added:  API: OpenGL v: 4.6 Mesa 23.3~git2309050600.31b5f5~oibaf~l (git-31b5f5a
    2023-09-05 lunar-oibaf-ppa) renderer: AMD Radeon RX 5500 XT (navi14 LLVM
    15.0.7 DRM 3.49 6.2.0-32-generic)
cheers
bob

**Comment by aaronliu0130** [comment]
Thanks for the additional information! Do you have an integrated GPU that may have taken over upscaling instead?

**Comment by ejektaflex** [comment]
Same issue here, Windows 11.

In my case, the conversion seems to fail if the folder that the file exists inside of uses Japanese characters. I've tried this with Roman and Japanese character sets, and it fails when in a folder with a Japanese title, but not with an English one.

```
⚙️ Getting saveImageAs from localStorage png
⚙️ Getting model from localStorage {"label":"General Photo (Real-ESRGAN)","value":"realesrgan-x4plus"}
⚙️ Setting gpuId to empty string
🔀 Setting model to 
🔄 Resetting image paths
🔀 Setting model to realesrgan-x4plus
🔄 Resetting image paths
⤵️ Dropped file:  {"type":"image/png","filePath":"あ\\abc.png","extension":"png"}
🖼 Setting image path:  C:\あ\abc.png
🗂 Setting output path:  C:\あ
🖼 imagePath:  C:\あ\abc.png
🔤 Extension:  png
🔄 Resetting Upscaled Image Path
🏁 UPSCAYL
🐞 BACKEND REPORTED:  📢 Upscayl Command:  -i,C:\あ\abc.png,-o,C:\あ\abc_upscayl_4x_realesrgan-x4plus.png,-s,4,-m,C:\Program Files\Upscayl\resources\models,-n,realesrgan-x4plus,,,-f,png
🐞 BACKEND REPORTED:  image upscayl:  [0 NVIDIA GeForce RTX 4080]  queueC=2[8]  queueG=0[16]  queueT=1[2]
[0 NVIDIA GeForce RTX 4080]  bugsbn1=0  bugbilz=0  bugcopc=0  bugihfa=0
[0 NVIDIA GeForce RTX 4080]  fp16-p/s/a=1/1/1  int8-p/s/a=1/1/1
[0 NVIDIA GeForce RTX 4080]  subgroup=32  basic=1  vote=1  ballot=1  shuffle=1

🚧 UPSCAYL_PROGRESS:  [0 NVIDIA GeForce RTX 4080]  queueC=2[8]  queueG=0[16]  queueT=1[2]
[0 NVIDIA GeForce RTX 4080]  bugsbn1=0  bugbilz=0  bugcopc=0  bugihfa=0
[0 NVIDIA GeForce RTX 4080]  fp16-p/s/a=1/1/1  int8-p/s/a=1/1/1
[0 NVIDIA GeForce RTX 4080]  subgroup=32  basic=1  vote=1  ballot=1  shuffle=1

🐞 BACKEND REPORTED:  image upscayl:  [1 Intel(R) UHD Graphics 630]  queueC=0[1]  queueG=0[1]  queueT=0[1]
[1 Intel(R) UHD Graphics 630]  bugsbn1=0  bugbilz=0  bugcopc=0  bugihfa=0
[1 Intel(R) UHD Graphics 630]  fp16-p/s/a=1/1/1  int8-p/s/a=1/1/1
[1 Intel(R) UHD Graphics 630]  subgroup=32  basic=1  vote=1  ballot=1  shuffle=1

🚧 UPSCAYL_PROGRESS:  [1 Intel(R) UHD Graphics 630]  queueC=0[1]  queueG=0[1]  queueT=0[1]
[1 Intel(R) UHD Graphics 630]  bugsbn1=0  bugbilz=0  bugcopc=0  bugihfa=0
[1 Intel(R) UHD Graphics 630]  fp16-p/s/a=1/1/1  int8-p/s/a=1/1/1
[1 Intel(R) UHD Graphics 630]  subgroup=32  basic=1  vote=1  ballot=1  shuffle=1

🐞 BACKEND REPORTED:  image upscayl:  0.00%

🚧 UPSCAYL_PROGRESS:  0.00%

🐞 BACKEND REPORTED:  image upscayl:  8.33%

🚧 UPSCAYL_PROGRESS:  8.33%

🐞 BACKEND REPORTED:  image upscayl:  16.67%

🚧 UPSCAYL_PROGRESS:  16.67%

🐞 BACKEND REPORTED:  image upscayl:  25.00%

🚧 UPSCAYL_PROGRESS:  25.00%

🐞 BACKEND REPORTED:  image upscayl:  33.33%

🚧 UPSCAYL_PROGRESS:  33.33%

🐞 BACKEND REPORTED:  image upscayl:  41.67%

🚧 UPSCAYL_PROGRESS:  41.67%

🐞 BACKEND REPORTED:  image upscayl:  50.00%

🚧 UPSCAYL_PROGRESS:  50.00%

🐞 BACKEND REPORTED:  image upscayl:  58.33%

🚧 UPSCAYL_PROGRESS:  58.33%

🐞 BACKEND REPORTED:  image upscayl:  66.67%

🚧 UPSCAYL_PROGRESS:  66.67%

🐞 BACKEND REPORTED:  image upscayl:  75.00%

🚧 UPSCAYL_PROGRESS:  75.00%

🐞 BACKEND REPORTED:  image upscayl:  83.33%

🚧 UPSCAYL_PROGRESS:  83.33%

🐞 BACKEND REPORTED:  image upscayl:  91.67%

🚧 UPSCAYL_PROGRESS:  91.67%

🐞 BACKEND REPORTED:  💯 Done upscaling
🐞 BACKEND REPORTED:  ♻ Scaling and converting now...
🐞 BACKEND REPORTED:  ❌ Error reading original image metadata Error: ENOENT: no such file or directory, open 'C:\あ\abc_upscayl_4x_realesrgan-x4plus.png'
🚧 UPSCAYL_PROGRESS:  Error: ENOENT: no such file or directory, open 'C:\あ\abc_upscayl_4x_realesrgan-x4plus.png'
⚙️ Getting saveImageAs from localStorage png
⚙️ Getting model from localStorage {"label":"General Photo (Real-ESRGAN)","value":"realesrgan-x4plus"}
⚙️ Setting gpuId to empty string
⚙️ Getting rememberOutputFolder from localStorage false
```

**Comment by aaronliu0130** [comment]
@ejektaflex That’s also a bug, but OP’s logs indicate that they selected a Roman directory. Could you file a new issue?

**Comment by tpfbnz** [comment]
Hi, I do have an integrated GPU, but it is not selected in UEFI ( for video) and its memory is allocated back to system. I don't know if computational programs can access it though.
Edit from inxi
System:
Kernel: 6.2.0-32-generic x86_64 bits: 64
    Desktop: Cinnamon 5.8.4 Distro: Linux Mint 21.2 Victoria
Machine:
  Type: Desktop Mobo: ASUSTeK model: TUF GAMING X570-PLUS (WI-FI) v: Rev X.0x
  CPU:
  Info: 6-core model: AMD Ryzen 5 5600G with Radeon Graphics bits: 64
 Graphics:
  Device-1: AMD Navi 14 [Radeon RX 5500/5500M / Pro 5500M] driver: amdgpu
    v: kernel
  Display: x11 server: X.Org v: 1.21.1.4 driver: X: loaded: amdgpu,ati
    unloaded: fbdev,modesetting,radeon,vesa gpu: amdgpu resolution: 3840x2160
  OpenGL: renderer: AMD Radeon RX 5500 XT (navi14 LLVM 15.0.7 DRM 3.49
  6.2.0-32-generic)
    v: 4.6 Mesa 23.0.4-0ubuntu1~22.04.1

**Comment by tpfbnz** [comment]
Hi, This log is similar but please note that the input file name was changed, presumably at arounfd "Done upscaling" and "Scaliing and converting"
Likely that each program segment handled file separately with doneup scaling adding to filename to generate output file name, then the next section assuming input file name is correct but using name still in the file select field but it has already been used to setup output file name and not cleared. Notice diff from 1st raspberry icon to the fourth raspberry icon in the list below.

I have created a suggestion in issues about this phenomenon

🐞 BACKEND REPORTED:  📢 Upscayl Command:  -i,/home/tpf/Pictures/329278.jpg,-o,/home/tpf/Pictures/2/329278_upscayl_4x_realesrgan-x4plus.png,-s,4,-m,/opt/Upscayl/resources/models,-n,realesrgan-x4plus,-g,0,-f,png
🐞 BACKEND REPORTED:  💯 Done upscaling
🐞 BACKEND REPORTED:  ♻ Scaling and converting now...
🐞 BACKEND REPORTED:  ❌ Error reading original image metadata Error: ENOENT: no such file or directory, open '/home/tpf/Pictures/2/329278_upscayl_4x_realesrgan-x4plus.png'
🚧 UPSCAYL_PROGRESS:  Error: ENOENT: no such file or directory, open '/home/tpf/Pictures/2/329278_upscayl_4x_realesrgan-x4plus.png'
cheers
bob

**Comment by aaronliu0130** [comment]
The output file name is supposed to be changed. We don't overwrite the input file.

**Comment by tpfbnz** [comment]
I feel I am being misunderstood. I apologise if I am wrong but something does not work and it precludes me from using Upsacyl at all because since upgrading it does not work for me. I am making a suggestion about the cause of this.

- Please compare carefully the top 🐞 BACKEND REPORTED:  above.  The input file name is  -i,/home/tpf/Pictures/329278.jpg,
- The 4th 🐞 BACKEND REPORTED:  says Error reading original image metadata, which I take to mean it is looking to read the input file seeking metadata.
- It reports it does not find "open '/home/tpf/Pictures/2/329278_upscayl_4x_realesrgan-x4plus.png' "
- I say that means it is looking for input file name it holds
-  but the utility part of the program looks at the wrong file name 
- because the filename processing section has been used to set the output file
-  and the input file name no longer exists there. 
- If my conjecture is correct, then a programming change is required to get the correct filename i.e. the input filename, which should be sent to the read utility so the metadata can be read and the processing to continue
I am sure the intention is to change output file name and that is done.
It is the timing that I suggest causes the processing to fail to complete.
I have opened #454 as a suggested method of repair.

**Comment by aaronliu0130** [comment]
We use real-esrgan to upscayl which immediately saves the file to filename_upscayl_modelname.png . After this, the upscayling is done. This happens in logs from "Upscayl Command" to percentage to "Done upscaling". In 2.7.5, we added #370 which (using JIMP which we plan to replace) compresses and converts the filename_upscayl_modelname.png to the desired extension. That is the "Scaling and converting now" step. It says "original" metadata because JIMP doesn't know we're upscaling; it only knows we're converting a file to a file with the same name, and it thinks the former is the "original file".

**Comment by NayamAmarshe** [comment]
We've replaced JIMP with Sharp.js

**Comment by NayamAmarshe** [comment]
@tpfbnz Could you please confirm if this issue has been fixed?

**Comment by tpfbnz** [comment]
Hi, sorry about delay, life! Who would have thought? :) 
Just installed new version, 2.8.1, and it works well producing results, both single and in batch.
But the  Upscayl  entry fields for input and output folders can appear misleading. I note that Upscayl accepted batch output folder details then added extra details to folder name and created a new folder along side the designated folder. Not a killer, but momentarily disconcerting when the processing finished and I looked in the folder I was expecting it to be in. 

Also , while testing it,  default assignment of output, overwriting input in Input directory with the output occurred. I did not notice it and increasingly large files were produced, until processing failed to finish. At least, that is my assumption of what happened.
[Upscayllogfilebob.txt](https://github.com/upscayl/upscayl/files/12680195/Upscayllogfilebob.txt)

I have included a log file of the last quick testing run done while writing this. I am pleased Upscayl works as well as it does in this release, I should find it fun to improve old photos.
Cheers
Bob
